src/pudl/output/ferc714.py

```diff
@@ -0,0 +1,400 @@
+            demand_annual = (
+                pd.merge(
+                    self.annualize(update=update),
+                    self.pudl_out.demand_hourly_pa_ferc714()
+                    .loc[:, ["report_date", "respondent_id_ferc714", "demand_mwh"]],
+                    how="left")
+                .groupby(["report_date", "respondent_id_ferc714"])
+                .agg({"demand_mwh": sum})
+                .rename(columns={"demand_mwh": "demand_annual_mwh"})
+                .reset_index()
+            )
+            # Merge respondent categorizations into the annual demand
+            self._demand_summary = (
+                pd.merge(demand_annual, self.categorize(update=update), how="left")
+                .pipe(
+                    pudl.helpers.convert_cols_dtypes,
+                    data_source="ferc714",
+                    name="FERC 714 Respondents Demand Summary",
+                )
+                .pipe(
+                    pudl.helpers.convert_cols_dtypes,
+                    data_source="eia",
+                    name="FERC 714 Respondents Demand Summary",
+                )
+            )
+        return self._demand_summary
+
+    def fipsify(self, update=False):
+        """
+        Annual respondents with the county FIPS IDs for their service territories.
+
+        Given the ``respondent_type`` associated with each respondent (either
+        ``utility`` or ``balancing_authority``) compile a list of counties that are part
+        of their service territory on an annual basis, and merge those into the
+        annualized respondent table. This results in a very long dataframe, since there
+        are thousands of counties and many of them are served by more than one entity.
+
+        Currently respondents categorized as ``utility`` will include any county that
+        appears in the ``service_territory_eia861`` table in association with that
+        utility ID in each year, while for ``balancing_authority`` respondents, some
+        counties can be excluded based on state (if ``self.limit_by_state==True``).
+        """
+        if update or self._fipsified is None:
+            categorized = self.categorize(update=update)
+            # Generate the BA:FIPS relation:
+            ba_counties = (
+                pudl.analysis.service_territory.get_balancing_authority_counties(
+                    ba_ids=categorized.balancing_authority_id_eia.unique(),
+                    st_eia861=self.pudl_out.service_territory_eia861(),
+                    ba_assn_eia861=self.pudl_out.balancing_authority_assn_eia861(),
+                    limit_by_state=self.limit_by_state
+                )
+            )
+            ba_fips = (
+                pd.merge(
+                    categorized.query("respondent_type=='balancing_authority'"),
+                    ba_counties,
+                    on=["report_date", "balancing_authority_id_eia"],
+                    how="left",
```

Comment:
are you making this `ba_fips` from `ba_counties` because you want every BA/report_date combo to be in the df? Maybe this seems confusing to me bc I assume `ba_counties` has fips codes in it so `ba_fips` seems weirdly named. But also it is not clear to me in general why this is necessary.

Reply:
I think I could just merge it back into the same `ba_counties` dataframe rather than creating a `ba_fips` here. Maybe that would be clearer. This is about merging the EIA 861 service territory information and the categorized FERC 714 respondent information together in one dataframe.

Reply:
Okay I got rid of these intermediary DFs and it's shorter and cleaner now.